### ISSUE TITLE
Evaluate GitHub labels

### ISSUE THREAD

**Issue by zackkrida** [WordPress/openverse#214]
## Assigned reviewers

- [x] @dhruvkb 
- [x] @AetherUnbound 

## Description

@dhruvkb identified some issues with our existing labels:

- emoji making labels harder to type
- priority labels being confusing (and somewhat arbitrary)
- labels not aligning directly with status columns in projects
- design not factored in status (considering their historically cross-repo nature)

Let's talk about these here, and make any additional issues as needed.

**Comment by zackkrida** [comment]
> emoji making labels harder to type

I like the emoji, but just because they are cute, not because they offer any functionality. I would be okay with removing them. We could continue to use them in our data model for labels; they may be helpful in other integrations; things like [overvue](https://github.com/sarayourfriend/openverse-overvue).

> priority labels being confusing (and somewhat arbitrary)

I don't personally find these confusing. Removing 'high' as a default for newly-created bugs has helped me keep track of critical and high bugs. I think `critical` almost always refers to something broken in production or something immediately blocking deployment, while high is things that impact or slow down or work, and could _become_ critical over time. `Low` makes sense as things that are nice to have, but not essential, and `medium` is a healthy default for in-progress and upcoming work that isn't tightly coupled to an approaching deadline.

> labels not aligning directly with status columns in projects

For me this is okay. 

> design not factored in status (considering their historically cross-repo nature)

I propose a new `aspect: design` label for this, that would be removed and replaced with another aspect once the design work is ready for development.

**Comment by panchovm** [comment]
Thanks for making this RFC.

> priority labels being confusing (and somewhat arbitrary)

I do find these labels confusing but likely because I tend to follow the project board's columns rather than filtering by priority in the list view. I would like what other folks think about this.

> design not factored in status (considering their historically cross-repo nature)

Regarding the design work, I think is necessary to make more visible the backlog, in progress, and done tasks. Much of the pending work is in personal lists and the community can not see the whole scope. On the other hand, sharing an idea and collecting feedback until reaching the final solution is not linear at all. 

Even when I like the idea of having one issue per problem that evolves from stage to stage, it seems necessary to have the design tasks separated from the dev work. This will keep the design request clear, it will ease the collaboration of new contributors, and it will allow us to trace the historical progress. We have also seen issues embracing several dev changes because a design proposal impacts many areas, whereas a separated design issue aids the creation of multiple and specific dev issues once the design reaches a certain approval level.

**Comment by AetherUnbound** [comment]
> emoji making labels harder to type

I must admit, I'm also fond of the emojis :octopus: When I'm labeling an issue with the GitHub CLI [(video)](https://asciinema.org/a/3SooBHRNLmnPllpe3FSRtbNH3), I use the emojis as quick markers when filtering. I think the biggest pain point with using them is when performing an issue search, which can be mitigated by 1) clicking the label in the issues page to filter down to issues with that label or 2) copy and pasting the contents of the label in the search bar. I'm likely to do the second one anyway if I'm typing out multiple labels for searches. So that to say I think they provide more benefits than drawbacks presently (the serotonin I get when seeing them is important, okay!! :grin:)

> priority labels being confusing (and somewhat arbitrary)
> labels not aligning directly with status columns in projects

I don't find either of these to be issues, personally.

> design not factored in status (considering their historically cross-repo nature)

I think this brings up another interesting point. For the catalog, the `aspect: a11y` and `aspect: interface` labels don't seem as relevant. Almost every issue we make is `aspect: text`, `aspect: dx`, or `aspect: code`. While I fully support an `aspect: design` label, I don't think it'll be very relevant to the catalog. Perhaps we can sync certain labels with only a subset of repositories?

**Comment by sarayourfriend** [comment]
I don't personally like the emojis but do not feel strongly about removing or adding them. I think they are cute, but they are also extra information that for people who do not rely on a screen reader is easy to ignore but presents redundant and potentially confusing or non-useful information for those relying on a screen reader. I do also find them annoying for typing issue filters but will admit that I could just bookmark the filter groups I like to use and deal with having to copy and paste them (using a mouse, which is annoying, potentially also annoying for users of accessible devices).

For the priority, I am fine with them in their current state. I agree with the suggestion raised recently (I am not remembering from who, sorry!) to add the justification for the priority in the issue. Perhaps this could be added to the issue template, "What priority do you think this is and why?".

**Comment by zackkrida** [comment]
@WordPress/openverse please add your feedback here next week if you haven't yet; thank you!

**Comment by dhruvkb** [comment]
I love emojis, they're cute and great for immediate visual cognition. Like I can look at the labels and quickly conclude what the issue is about and what the priority is based on the emoji. 

But as it stands, GitHub makes us put these emojis in the title, which is super bad. It basically makes everything we do with emoji harder (if you are not visually impaired) and frustrating/annoying (if you are) and so I'd just be fine if we got rid of them. With a judicious use of colors, we might just get a decent visual hit from the plain-text labels too!

As for repo-specific labels, the sync action does not remove any extra labels, it just makes sure the given base set exists everywhere. So we're free to add as many repo-specific labels, they'll just be left untouched.

**Comment by zackkrida** [comment]
> So we're free to add as many repo-specific labels, they'll just be left untouched.

And in fact, we already have some in the catalog, for example the `☁️ provider: *` labels.

**Comment by AetherUnbound** [comment]
> And in fact, we already have some in the catalog, for example the `☁️ provider: *` labels.

Of course 🤦🏼‍♀️ Maybe we should just drop #216 and add those labels only to the infra repo?

**Comment by dhruvkb** [comment]
@AetherUnbound considering that they'll exclusively be used in the infra repo makes me think adding it there would be best. Or we could add them universally and never use them, no harm either way.

**Comment by krysal** [comment]
An emojis lover here too 🙋🏻‍♀️ And yet I must admit that they hinder more than they contribute to the labels as they are currently used. My main problem is when searching by a label in some part of the GitHub UI where they are not listed or hinted, it's frustrating having to guess what the exact one was or having to go copy it when I know the label text, so maybe removing them would be the best.

Regarding labels for design work, which ones were you suggesting Francisco? `aspect: design` and something else? I can't remember but I'll likely support it, makes sense to me to create separate issues for design tasks with their relevant labels 👍

**Comment by panchovm** [comment]
Exactly @krysal, replacing `needs design` with `aspect: design`. But most importantly, creating new issues for design requirements and, once reaching an approval level, creating one or several issues labeled with `aspect` to begin its development. We can also use the `priority` label for design issues to set the focus work.

**Comment by zackkrida** [comment]
@panchovm I've created PR #220 for the new `aspect: design` label; please review.

**Comment by obulat** [comment]
I am a little confused about the `needs design` label, @panchovm.
This is how I understand the [comment above](https://github.com/WordPress/openverse/issues/214#issuecomment-1099649060):
If an issue is opened, and we realize that it needs design work before being implemented in code, we add `aspect:design` label to it. Then, after you finish the designs, there are two scenarios: 
1. you comment saying that the design is ready and link to the Figma mockup, and someone else opens a new issue, copying some information from the original one, but this time instead of adding the `aspect: design` label, we add `aspect: interface`, or some other code-related `aspect`.
2. You open the new issue yourself (again, with a different `aspect` label), and close the original one as finished.

This way, as soon as design work is ready, the issue with `aspect: design` is closed as done. And we do not use `needs design` label at all.

Do I understand it correctly, @panchovm?

**Comment by sarayourfriend** [comment]
FWIW that is has also been my understanding of the intended process.

**Comment by panchovm** [comment]
I was thinking of the following flow:
1. In any ticket, if the discussion concludes that we need a design solution, anyone can add the `needs design` requesting a design solution. In this way, anyone wanting to contribute with design can find what design needs openverse has.
2. If I decide to propose a design for an issue labeled with `needs design`, I create a new issue mentioning the ticket and add the `aspect: design` to the new to share and iterate on the design proposal.
3. After discussing and deciding to develop the design proposal in the `aspect: design` issue, we create a new dev issue mentioning the issue so we can track the design issues and know where this implementation is coming from.

<img width="700" alt="CleanShot 2022-05-06 at 15 26 41@2x" src="https://user-images.githubusercontent.com/895819/167204728-d847856a-cf1a-47ce-8135-ce33b396fbb8.png">

**Comment by obulat** [comment]
> I was thinking of the following flow:

Thank you for your beautiful reply, @panchovm! A couple more points to clarify:
Do we close the first `needs design` issue after you create the `aspect: design` issue with the design proposal? Or do we just remove the `needs design` label?
I guess we should close it because we are going to open a new development issue after the design proposal has been accepted (i.e., issue #2 is closed).

**Comment by AetherUnbound** [comment]
I think we've reached general consensus here, so I'm closing this issue. If you have more to add though, please do so!
